src/migrations/1722004753128-write-offs.ts

```diff
@@ -0,0 +1,120 @@
+import { getAdminPermissions } from './1720624912260-database-rbac';
+
+export class WriteOffs1722004753128 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.createTable(new Table({
+      name: 'write_off',
+      columns: [
+        {
+          name: 'createdAt',
+          type: 'datetime(6)',
+          default: 'current_timestamp',
+          isNullable: false,
+        },
+        {
+          name: 'updatedAt',
+          type: 'datetime(6)',
+          default: 'current_timestamp',
+          onUpdate: 'current_timestamp',
+          isNullable: false,
+        },
+        {
+          name: 'version',
+          type: 'integer',
+          isNullable: false,
+        },
+        {
+          name: 'id',
+          type: 'integer',
+          isPrimary: true,
+          isGenerated: true,
+          generationStrategy: 'increment',
+        },
+        {
+          name: 'transferId',
+          type: 'integer',
+          isNullable: true,
+        },
+        {
+          name: 'amount',
+          type: 'integer',
+          isNullable: false,
+        },
+        {
+          name: 'toId',
+          type: 'integer',
+          isNullable: false,
+        },
+      ],
+    }), true);
+
+    await queryRunner.createForeignKey('write_off', new TableForeignKey({
+      name: 'FK_write_off_transferId',
+      columnNames: ['transferId'],
+      referencedColumnNames: ['id'],
+      referencedTableName: 'transfer',
+      onDelete: 'CASCADE',
+    }));
+
+    await queryRunner.createForeignKey('write_off', new TableForeignKey({
+      name: 'FK_write_off_toId',
+      columnNames: ['toId'],
+      referencedColumnNames: ['id'],
+      referencedTableName: 'user',
+      onDelete: 'CASCADE',
+    }));
+
+    const roleRepo = queryRunner.manager.getRepository(Role);
+    const permissionRepo = queryRunner.manager.getRepository(Permission);
+
+    await roleRepo.findOne({ where: { name: 'SudoSOS - BAC PM' } }).then(async (role) => {
+      if (!role) return;
+      role.permissions = [...role.permissions, ...await permissionRepo.save({ ...getAdminPermissions(role, 'Transaction') })];
+      await roleRepo.save(role);
+    });
+
+    await roleRepo.findOne({ where: { name: 'SudoSOS - Audit' } }).then(async (role) => {
+      if (!role) return;
+      role.permissions = [...role.permissions, ...await permissionRepo.save([
+        { roleId: role.id, role, entity: 'WriteOff', action: 'get', relation: 'all', attributes: ['*'] },
+      ])];
+      await roleRepo.save(role);
+    });
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.dropTable('write_off');
+    const roleRepo = queryRunner.manager.getRepository(Role);
+    await roleRepo.findOne({ where: { name: 'SudoSOS - BAC PM' } }).then(async (role) => {
+      if (!role) return;
+      role.permissions = role.permissions.filter((p) => p.entity !== 'WriteOff');
```

Comment:
You can also do `queryRunner.getRepository(Permission).remove({ entity: 'WriteOff' })`, which would delete all permissions that have that particular entity.